File: src/Java/pipeline-examples/src/main/java/com/pipeline/examples/Example01_TextClean.java
package com.pipeline.examples;

import com.pipeline.core.Pipeline;
import com.pipeline.examples.steps.TextSteps;

public final class Example01_TextClean {
  private Example01_TextClean() {}

  public static void run() {
    Pipeline<String> pipeline = new Pipeline<String>("ex01", /*shortCircuitOnException=*/false)
        .addAction(TextSteps::strip)
        .addAction(TextSteps::normalizeWhitespace)
        .addAction(TextSteps::truncateAt280);

    String input = "  Hello   <b>World</b>  ";
    String outputValue = pipeline.run(input).context();
    System.out.println("[ex01] => " + outputValue);
  }
}

File: src/CSharp/pipeline_services_examples/Example01TextClean.cs
using System;

using PipelineServices.Core;

namespace PipelineServices.Examples;

public static class Example01TextClean
{
    public static void Run()
    {
        Pipeline<string> pipeline = new Pipeline<string>("example01_text_clean", shortCircuitOnException: true);
        pipeline.AddAction(TextActions.Strip);
        pipeline.AddAction(TextActions.NormalizeWhitespace);
        pipeline.AddAction("truncate", TextActions.TruncateAt280);

        PipelineResult<string> result = pipeline.Run("  Hello   World  ");
        Console.WriteLine("output=" + result.Context);
        Console.WriteLine("shortCircuited=" + result.ShortCircuited);
        Console.WriteLine("errors=" + result.Errors.Count);
    }
}
